server/src/main/java/org/elasticsearch/search/aggregations/metrics/MaxAggregator.java

```diff
@@ -186,7 +186,10 @@ public void visit(int docID, byte[] packedValue) {
                 if (liveDocs.get(docID)) {
                     // we need to collect all values in this leaf (the sort is ascending) where
                     // the last live doc is guaranteed to contain the max value for the segment.
-                    result[0] = converter.apply(packedValue);
+                    if (result[0] == null) {
+                        result[0] = new byte[packedValue.length];
```

Comment:
I hadn't thought about this before, @iverase do you know if the `packedValue` is guaranteed to have same length for all values in the tree?  Or is it a variable packing scheme?

Reply:
:+1: ta